lib/src/modules/network_logger.dart

```diff
@@ -17,8 +19,21 @@ class NetworkLogger {
     _host = host;
   }
 
+  /// @nodoc
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void $setManager(NetworkManager manager) {
+    _manager = manager;
+  }
+
+  static void obfuscateLog(ObfuscateLogCallback callback) {
+    _manager.setObfuscateLogCallback(callback);
+  }
```

Comment:
I suggest that we add docs with example to this method, what do you think? (a doc with an example for usage)
```dart
/// Registers a callback to obfuscate sensitive network log data.
///
/// Use this method to define a callback that will modify or obscure
/// sensitive information within network logs before they are recorded
/// or displayed.
///
/// The [callback] function takes a [NetworkData] object and should
/// return a modified [NetworkData] object.
///
/// Example:
/// 
/// NetworkLogger.obfuscateLog((data) {
///   // Implement custom logic to obfuscate sensitive data here.
///   // Modify 'data' as needed and return it.
/// });
/// 
static void obfuscateLog(ObfuscateLogCallback callback) {
  _manager.setObfuscateLogCallback(callback);
}
```

Reply:
@a7medev, I've made the updated mentioned here with a slight adjustment, could you please check them?

Reply:
Thanks a lot, @abdelhamid-f-nasser, once again! I've made small changes and moved it to `NetworkLogger` as well 🙏🏼